File: Test projects/.NET6.0/WinFormsTestAppNet6CSharp/Form1.cs
using DDDSoft.Windows.Winforms.Extensions;
using DDDSoft.Windows.Winforms.Navigation;
using WinFormsTestAppNet6CSharp.Forms;

namespace WinFormsTestAppNet6CSharp
{
    public partial class Form1 : Form
    {
        private readonly IFormNavigator _formNavigator;

        public Form1(IFormNavigator formNavigator)
        {
            _formNavigator = formNavigator;

            InitializeComponent();
        }

        private void ButtonOpenTestForm1_Click(object sender, System.EventArgs e)
        {
            _formNavigator.ShowForm<TestForm>();
        }

        private void ButtonOpenTestForm2_Click(object sender, System.EventArgs e)
        {
            DialogResult result = _formNavigator.ShowDialog<DialogForm>();
        }
    }
}

File: Test projects/.NET48/WinformsTestAppNet48Vb/Form1.vb
Imports DDDSoft.Windows.Winforms.Navigation
Imports DDDSoft.Windows.Winforms.Extensions

Public Class Form1

    Private ReadOnly _formNavigator As IFormNavigator

    Public Sub New(formNavigator As IFormNavigator)
        _formNavigator = formNavigator

        InitializeComponent()

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        _formNavigator.ShowForm(Of TestForm)()

    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click

        Dim result As DialogResult = _formNavigator.ShowDialog(Of DialogForm)()

    End Sub

End Class
